scrapy/downloadermiddlewares/httpcompression.py

```diff
@@ -44,10 +48,11 @@ def process_response(self, request, response, spider):
                     # force recalculating the encoding until we make sure the
                     # responsetypes guessing is reliable
                     kwargs['encoding'] = None
+                if self.keep_encoding_header:
+                    kwargs['flags'] = [encoding]
                 response = response.replace(**kwargs)
                 if not content_encoding:
                     del response.headers['Content-Encoding']
```

Comment:
Shouldn’t this *not* be done if `self.keep_encoding_header` is `True`?

Reply:
Ar first I had it just skip that but there are a few tests that assert 'Content-Encoding' is None and also there was what @kmike asked in the [issue comments](https://github.com/scrapy/scrapy/issues/1988#issuecomment-219405456) about placing it in the response.flags. So with those two things decided to allow the header to be removed but have the encoding placed in the response.flags to signal it was compressed before processing. ( I believe that's what flags are used for if I'm misunderstanding their use as signals about the what has been done or what is wanted with a response?)

Reply:
OOh okay I see what you mean. That make sense, I wasn't aware enough of how the headers where used later on and the particulars of what flags where for. Especially them being checked before hand, can you think of a good example of that in the code, I'm having trouble?